pkgs/by-name/co/code-cursor/package.nix

```
@@ -54,41 +70,77 @@ let
     runHook postInstall
   '';
 
+  darwinInstallPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/Applications
+    cp -r *.app $out/Applications/
+
+    mkdir -p "$out/bin"
+    cat << EOF > "$out/bin/cursor"
+    open -na '$APP_DIR' --args "\$@"
+    EOF
+    chmod +x "$out/bin/cursor"
+    runHook postInstall
+  '';
+
 in
 stdenvNoCC.mkDerivation {
   inherit pname version;
 
-  src = wrappedAppImage;
+  src = if buildPlatform.isLinux then wrappedAppImage else null;
+  
+  dontBuild = buildPlatform.isDarwin;
 
-  nativeBuildInputs = [ makeWrapper ];
+  nativeBuildInputs = [ makeWrapper ] ++ lib.optional buildPlatform.isDarwin undmg;
 
-  installPhase = appimageInstall;
+  srcRoot = .;
```

```suggestion
  sourceRoot = ".";
```

Sorry, ID10T error on my part.